lib/database_cleaner/base.rb

```diff
@@ -48,8 +45,7 @@ def strategy
     attr_reader :orm
 
     def orm=(desired_orm)
-      @orm = (desired_orm || :autodetect).to_sym
-      @orm = @orm_autodetector.orm if @orm == :autodetect
+      @orm = desired_orm && desired_orm.to_sym
```

Comment:
@botandrose I don't understand this line. Wouldn't this be the same?

```suggestion
      @orm = desired_orm.to_sym
```

Reply:
Nil raises an error if you do that.

Reply:
Hmm yeah. I've added a test case to the spec to ensure that passing nil doesn't cause it to crash.

Reply:
Although, now I'm thinking it should raise `NoORMDetected` like `DatabaseCleaner[nil]` does. What do you guys think?

Reply:
Probably should rename that exception, too, while we're at it, now that there's no more orm detection!

Reply:
I mean I think its far to raise an `ArgumentError` here at least...

Reply:
Yeah, good call. That sounds like the right move to me. Replace NoORMDetected with ArgumentError, and raise it in both cases.

Reply:
Digging into this issue, I began to wonder if this is an opportunity to expose NullStrategy? For example:

```ruby
DatabaseCleaner[nil].clean # => Clean with NullStrategy
DatabaseCleaner.orm = nil # => Set to NullStrategy
DatabaseCleaner.strategy = nil # => Set to NullStrategy
```

This is getting a little out of scope of this PR, but what do y'all think?

Reply:
I think the likely hood of someone doing that intentionally is lower than someone making a mistake and passing nil unintentionally, you'd still want an error or a warning somewhere even if you did that.